### PR TITLE
Added Korean translation

### PR DESCRIPTION
**Pull request by KANGPUNGYUN**
### Description

I translated Euclidean algorithm and Magic square explanation into Korean. And I added missing value in _magic_square.md_.

### Checklist

- [x] A description of the changes in this PR is mentioned above.
- [x] All the new and existing checks pass.
- [x] No plagiarized, duplicated, or repetitive documentation that has been directly copied from another source.
- [x] If it's a new explanation, it contains solid, understandable, and accessible information.
- [x] I have read the whole [contributing guidelines](https://github.com/TheAlgorithms/Algorithms-Explanation/blob/master/CONTRIBUTING.md) and agree to the [Code of Conduct](https://github.com/TheAlgorithms/.github/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

**Review by appgurueu — APPROVED** [review]
Looks fine (again only roughly compared using Google Translate).

Perhaps you should propagate the fix in the english original file (missing number in magic square) to all translations.

**Comment by KANGPUNGYUN** [comment]
You're right. I'll fix it right away!:)

**Comment by KANGPUNGYUN** [comment]
There are no file written in another language.

**Review by appgurueu — APPROVED** [review]

File: ko/기초 수학/마방진.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# 참조
+
+## 웹사이트:-
+
+- [Byjus](https://byjus.com/maths/magic-square/)
+- [geeksforgeeks](https://www.geeksforgeeks.org/magic-square/)
+
+## 유투브:-
+
+- [영상 "Why is there only one 3x3 magic square?"](https://www.bing.com/videos/search?q=magic+square&&view=detail&mid=26BE595B719B8B532E5126BE595B719B8B532E51&&FORM=VRDGAR&ru=%2Fvideos%2Fsearch%3Fq%3Dmagic%2Bsquare%26FORM%3DHDRSC3)
```
Comment (Panquesito7) [inline ko/기초 수학/마방진.md:72]: Might want to add a page to the The Algorithms' page explanation of this algorithm: https://the-algorithms.com/algorithm/magic-square

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: ko/기초 수학/마방진.md (hunk 1 above)
Comment (KANGPUNGYUN) [reply ko/기초 수학/마방진.md]: Oh, I got it. I just edited it!

File: ko/기초 수학/유클리드 알고리즘.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+## 활용
+
+- 분수 단축하기
+- 최소공배수 구하기
+- 두 수가 서로소인지 효율적으로 확인하기 (예: RSA 암호화 시스템에 필요)
+
+## 참고자료
+
+- [위키피디아 "유클리드 호제법" 항목](https://ko.wikipedia.org/wiki/%EC%9C%A0%ED%81%B4%EB%A6%AC%EB%93%9C_%ED%98%B8%EC%A0%9C%EB%B2%95)
```
Comment (Panquesito7) [inline ko/기초 수학/유클리드 알고리즘.md:88]: Might want to do the same here, with the link to the implementation.

Comment (KANGPUNGYUN) [reply ko/기초 수학/유클리드 알고리즘.md:88]: I just edited it!

**Review by appgurueu — APPROVED** [review]

**Review by madhavgaba — APPROVED** [review]

File: ko/기초 수학/유클리드 알고리즘.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+## 구현
+
+실제 실행에서 속도를 높이기 위해 반복해서 뺄셈하는 대신 나머지 연산이 사용된다:
+$b$는 $a >= b$만큼 $a$에서 여러 번 반복해서 뺄 수 있다.
+이러한 뺄셈 후에는 $b$로 나눌 때 $a$의 나머지만 남게 된다.
+
+간단한 Lua 구현은 다음과 같다:
+
+```lua
```
Comment (Panquesito7) [inline ko/기초 수학/유클리드 알고리즘.md:36]: Pseudocode would be better here, IMO.
What do you folks think about this?

Comment (appgurueu) [reply ko/기초 수학/유클리드 알고리즘.md:36]: The original file uses Lua; changing that is not in the scope of this translation PR. I've opened [a discussion](https://github.com/TheAlgorithms/Algorithms-Explanation/discussions/215) for this.

**Review by Panquesito7 — APPROVED** [review]
Thanks! 🚀

**Pull request merged**